### PR TITLE
Abandon camp NPC talk and bulletin board mission

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

```SUMMARY: Features "Abandon camp NPC talk and bulletin board mission"```

#### Purpose of change
Allow camps to be abandoned, all the buildings remain, and its much the same as just walking away, but some people may prefer it for neatness sake.

It also removes the camp from the overmapbuffer and submap, so a new camp could then be built nearby, perhaps in a slightly more advantageous position.

#### Describe the solution
add talk function and camp mission to abandon it, recall all workers, reassign all camp-assigned workers.

#### Describe alternatives you've considered
N/A
#### Testing
Talked to NPC at camp, asked them to abandon camp, it disappeared off hte overmap, and the bulletin board was not functional, and workers were recalled.

Tried the altenrative method of activcating the bulletin board mission to abandon camp, and the same happened.

#### Additional context
This is kinda pointless imo, if you dont want the camp there, just like .. walk away and forget about it, but a few people now have expressed the desire for this feature, so here it is :P

### CODE REVIEW

**Comment by Fris0uman** [comment]
What happens if you try to make a new camp on site of the old one?

**Comment by davidpwbrown** [comment]
> What happens if you try to make a new camp on site of the old one?

The camp overmap terrain is stil llisted as "camp survey" , I'm not sure a memory is kept of what overmap terrain it was before. 

But - now if they make a new camp, there will no longer be the thing stopping them of " there is already another camp nearby ".

All the buildings will still be there, depending on how upgraded the camp was, and if this overlaps with a new camp, the new camp will likely jsut overwrite and merge with th eold buildings, same as if you build a construction then upgrade your camp where the construction is.

I.e its not ideal, but if people really wanna move their camp next door, and make a new one, or neaten up their menus and maps and stuff by removing something unwanted, then they can now, it wont exist as a basecamp structure in the data anymore.

File: src/faction_camp.cpp [hunk 1]
```diff
@@ -1561,6 +1567,27 @@ void basecamp::start_upgrade( const std::string &bldg, const point &dir,
...
+        random_guy = guy;
+        const std::string return_msg = _( "responds to the emergency recall…" );
+        finish_return( *guy, false, return_msg, "menial", 0, true );
+    }
+    for( npc_ptr &guy : get_npcs_assigned() ) {
+        talk_function::stop_guard( *guy );
+    }
+    overmap_buffer.remove_camp( *this );
+    g->m.remove_submap_camp( random_guy->pos() );
+    add_msg( m_info, _( "You abandon %s" ), name );
```
Comment (Night-Pryanik) [inline src/faction_camp.cpp]:
```suggestion
    add_msg( m_info, _( "You abandon %s." ), name );
```

File: src/faction_camp.cpp [hunk 2]
```diff
@@ -1266,6 +1266,9 @@ void basecamp::get_available_missions( mission_data &mission_key )
                                   "\n\nRisk: None\n"
                                   "Time: Ongoing" ) );
         mission_key.add( "Assign Jobs", _( "Assign Jobs" ), entry );
+        entry = string_format( _( "Notes:\n"
```
Comment (BevapDin) [inline src/faction_camp.cpp]: Why is there a call to `string_format` without formatting parameters?
```suggestion
        entry = _( "Notes:\nAbandon this camp" );
```

File: src/npctalk_funcs.cpp [hunk 3]
```diff
@@ -346,6 +346,16 @@ void talk_function::assign_guard( npc &p )
     p.set_omt_destination();
 }
 
+void talk_function::abandon_camp( npc &p )
+{
+    ( void )p;
```
Comment (BevapDin) [inline src/npctalk_funcs.cpp]:
```suggestion
```
Why even have this?

Comment (davidpwbrown) [reply src/npctalk_funcs.cpp]: Used to be required before I changed my mind on how to structure the abandon camp function.

File: src/faction_camp.cpp [hunk 4]
```diff
@@ -1266,6 +1266,9 @@ void basecamp::get_available_missions( mission_data &mission_key )
                                   "\n\nRisk: None\n"
                                   "Time: Ongoing" ) );
         mission_key.add( "Assign Jobs", _( "Assign Jobs" ), entry );
+        entry = _( "Notes:\nAbandon this camp" );
+                                  "Abandon this camp" ) );
```
Comment (anothersimulacrum) [inline src/faction_camp.cpp]:
```suggestion
```

**Pull request merged**